src/photon/photon_scheduler.c

```diff
@@ -303,16 +352,33 @@ void process_message(event_loop *loop,
   } break;
   case GET_TASK: {
     /* Update the task table with the completed task. */
-    if (state->db != NULL && worker->task_in_progress != NULL) {
-      task_set_state(worker->task_in_progress, TASK_STATUS_DONE);
-      task_table_update(state->db, worker->task_in_progress,
-                        (retry_info *) &photon_retry, NULL, NULL);
-      /* The call to task_table_update takes ownership of the task_in_progress,
-       * so we set the pointer to NULL so it is not used. */
-      worker->task_in_progress = NULL;
-    } else if (worker->task_in_progress != NULL) {
-      free_task(worker->task_in_progress);
-      worker->task_in_progress = NULL;
+    task *task_in_progress = worker->task_in_progress;
+    task_spec *spec =
```

Comment:
Can you move this line into the if-block below? This is just personal preference, but I always feel that ternary operators are harder to read.